bundles/NewsletterBundle/config/pimcore/default.yaml

```diff
@@ -0,0 +1,39 @@
+pimcore_newsletter:
+    source_adapters:
+        defaultAdapter: pimcore_newsletter.document.newsletter.factory.default
+        csvList: pimcore_newsletter.document.newsletter.factory.csv
+#    use_specific: false # set true to use the following options for newsletter delivery
+#    sender:
+#        name: ''
+#        email: ''
+#    return:
+#        name: ''
+#        email: ''
+
+framework:
+    mailer:
+        enabled: true
+        transports:
+            pimcore_newsletter: native://default
+    messenger:
+        routing:
+            'Pimcore\Bundle\NewsletterBundle\Messenger\SendNewsletterMessage': pimcore_core
+
+pimcore:
+    objects:
+        class_definitions:
+            data:
+                map:
+                    newsletterActive: \Pimcore\Bundle\NewsletterBundle\Model\DataObject\ClassDefinition\Data\NewsletterActive
+                    newsletterConfirmed: \Pimcore\Bundle\NewsletterBundle\Model\DataObject\ClassDefinition\Data\NewsletterConfirmed
```

Comment:
Yes, Marco and I talked already talked about this. And even the classes rebuild fails at the moment, because the old class is loaded while the rebuild so there is a declaration error, but we're working on a solution for this

Reply:
marking as resolved: waiting for the general approach by Marco that will retroactively fix these cases

Reply:
We are trying to resolve with with an object class resolver like for the assets and documents, instead of calling the `model.factory` to load the classes for the phpdoc param and return types. 

Atm the classes rebuild fails.

Reply:
In case of Newsletter data types, it is not only class-rebuild but customer object is not loaded at all.

Reply:
i've tested and approved https://github.com/pimcore/compatibility-bridge-v10/pull/5, will merge if there's any better alternative solution is coming up.

Reply:
compatibility bridge should be optional for only migration from v10.